service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java

```diff
@@ -72,14 +83,64 @@ public void run() {
+                            // if there are request-level updates then they must be published to the
+                            // CMO_REQUEST_METADATA_UPDATE topic as well
+                            // regardless of which level the updates occur, the updates still need
+                            // to be persisted to the database
+
+                            if (requestService.requestHasUpdates(existingRequest, request)) {
+                                // if request-level metadata updates exist then persist and publish updates
+                                if (requestService.requestHasMetadataUpdates(existingRequest, request)) {
+                                    // persist request-level metadata updates to database
+                                    // existingRequest.updateRequestMetadata(request)
+                                    // requestService.updateRequestMetadata(existingRequest);
+
+                                    LOG.info("Publishing Request-level Metadata updates "
+                                            + "to CMO_REQUEST_METADATA_UPDATE");
+                                    // publish request-level metadata history to CMO_REQUEST_UPDATE_TOPIC ..
+                                    //maybe pass to new handler?
+                                    //messagingGateway.publish(existingRequest.getRequestId(),
+                                    //        CMO_REQUEST_UPDATE_TOPIC,
+                                    //        mapper.writeValueAsString(
+                                    //              existingRequest.getRequestMetadataList()));
+                                }
+
+                                // only publish sample-level metadata history to CMO_SAMPLE_METADATA_UPDATE
+                                //  for samples containing updates in the new request json received
+                                List<MetaDbSample> updatedSamples =
+                                        requestService.getRequestSamplesWithUpdates(request);
+                                if (!updatedSamples.isEmpty()) {
+                                    LOG.info("Publishing Sample-level Metadata updates to "
+                                            + "CMO_SAMPLE_METADATA_UPDATE for " + updatedSamples.size()
+                                            + " samples.");
+                                    // maybe pass along to a diff message handler?
+                                    for (MetaDbSample sample : updatedSamples) {
+                                        // persist sample-level metadata updates to database
+                                        // sampleService.updateSampleMetadata(sample)
+                                        //messagingGateway.publish(CMO_SAMPLE_UPDATE_TOPIC,
+                                        //        mapper.writeValueAsString(sample.getSampleMetadataList()));
+                                    }
```

Comment:
Need to confirm: Should we still publish the request json RECEIVED to CMO_NEW_REQUEST for the consistency checker?